### PR TITLE
config: enforce stricter validation rules and tune default config

### PR DESCRIPTION
**Pull request by jiangliu**
Enforce stricter validation rules and tune default configuration in misc/snapshotter/config.toml

### CODE REVIEW

File: misc/snapshotter/config.toml [hunk 1]
```diff
@@ -21,15 +21,15 @@ daemon_cpu_profile_duration_secs = 5
...
 nydusd_path = "/usr/local/bin/nydusd"
 nydusimage_path = "/usr/local/bin/nydus-image"
 # fusedev or fscache
 fs_driver = "fusedev"
 # How to process when daemon dies: "none", "restart" or "failover"
 recover_policy = "restart"
-# Specify a configuration file for nydusd
-nydusd_config = "/etc/nydus/nydusd-config.fusedev.json"
-# The fuse or fscache IO working threads started by nydusd
+# Nydusd worker thread number to handle FUSE or fscache requests, [0-1024]
```
Comment (imeoer) [inline misc/snapshotter/config.toml:32]: May be `[1-1024]`.

Comment (jiangliu) [reply misc/snapshotter/config.toml:32]: 0 means default, let nydusd to choose worker thread number:)

File: misc/snapshotter/config.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version = 1
 root = "/var/lib/containerd-nydus"
 # The snapshotter's GRPC server socket, containerd will connect to plugin on this socket
 address = "/run/containerd-nydus/containerd-nydus-grpc.sock"
-daemon_mode = "multiple"
+daemon_mode = "dedicated"
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply misc/snapshotter/config.toml:6]: Sorry, the mode is introduced by other pending patches:)

[the remark this answers is not in the document]
Comment (changweige) [reply misc/snapshotter/config.toml:6]: In fact, I like the name `dedicated` 🤗

File: misc/snapshotter/config.toml (hunk 1 above)
Comment (imeoer) [reply misc/snapshotter/config.toml:32]: Ok! May be we should explain the zero value here.

Comment (jiangliu) [reply misc/snapshotter/config.toml]: added a comment for it.

File: misc/snapshotter/config.toml [hunk 3]
```diff
@@ -65,8 +66,8 @@ enable_kubeconfig_keychain = false
 kubeconfig_path = ""
 # Fetch the private registry auth as CRI image service proxy
 enable_cri_keychain = false
-# the target image service when using image proxy
-image_service_address = ""
+# the target image service when using image proxy, default: /run/containerd/containerd.sock
```
Comment (changweige) [inline misc/snapshotter/config.toml:69]: Does snapshotter really set the default value `/run/containerd/containerd.sock` ?

Comment (jiangliu) [reply misc/snapshotter/config.toml:69]: You are right, fixed

**Review by changweige — APPROVED** [review]
LGTM

**Review by imeoer — APPROVED** [review]

**Pull request merged**